cnf-certification-test/platform/nodetainted/nodetainted.go

```diff
@@ -29,22 +28,35 @@ import (
 
 // NodeTainted holds information about tainted nodes.
 type NodeTainted struct {
-	timeout      time.Duration
-	ClientHolder *clientsholder.ClientsHolder
-	OCPContext   clientsholder.Context
+	ClientHolder      *clientsholder.ClientsHolder
+	OCPContext        clientsholder.Context
+	unitTestingResult bool
 }
 
-// NewNodeTainted creates a new NodeTainted tnf.Test.
-func NewNodeTaintedTester(timeout time.Duration, client *clientsholder.ClientsHolder, ctx clientsholder.Context) *NodeTainted {
+//go:generate moq -out nodetainted_moq.go . TaintedFuncs
+type TaintedFuncs interface {
+	runCommand(cmd string, ctx clientsholder.Context) (string, error)
+	GetKernelTaintInfo(ctx clientsholder.Context) (string, error)
+	GetModulesFromNode(ctx clientsholder.Context) []string
+	ModuleInTree(moduleName string, ctx clientsholder.Context) bool
+	GetOutOfTreeModules(modules []string, ctx clientsholder.Context) []string
+	SetTestingResult(result bool)
```

Comment:
In the end, moq is summarizing all (most) ins/outs of a function and collects it in an interface so that it can be extracted out of context and tested independently. I would put all the required functions inside the interface, including the Ginkgo ones. This way, no need for the SetTestingResult function. I don't like that we have conditional unit test code in the runtime since now the unit test is part of the main code as well (so it cannot test itself). Also make getting the env part of the interface